packages/core/src/__tests__/state-manager.test.ts

```diff
@@ -145,6 +145,32 @@ test('handleTip', async () => {
   await ceramic2.close();
 });
 
+test('handleTip for commit already in log', async () => {
+  const newContent = {foo: 'bar'}
+  const stream1 = await TileDocument.create<any>(ceramic, INITIAL_CONTENT, null, { anchor:false });
+  await stream1.update(newContent, null, { anchor: false })
+
+  const ceramic2 = await createCeramic(ipfs);
+  const retrieveCommitSpy = jest.spyOn(ceramic2.dispatcher, 'retrieveCommit');
+
+  const stream2 = await ceramic2.loadStream<TileDocument>(stream1.id, { syncTimeoutSeconds:0 });
+  const streamState2 = await ceramic2.repository.load(stream2.id, {});
+
+
+  retrieveCommitSpy.mockClear()
+  await (ceramic2.repository.stateManager as any)._handleTip(streamState2, stream1.state.log[1].cid);
+
+  expect(streamState2.state).toEqual(stream1.state);
+  expect(retrieveCommitSpy).toBeCalledTimes(7) // TODO(1421): This should be 2!
```

Comment:
this is pretty scary. Filed https://github.com/ceramicnetwork/js-ceramic/issues/1421 to clean this up